btc-deposit/src/main/kotlin/notary/btc/listener/BitcoinBlockChainListener.kt

```diff
@@ -0,0 +1,51 @@
+package notary.btc.listener
+
+import io.reactivex.ObservableEmitter
+import mu.KLogging
+import org.bitcoinj.core.Block
+import org.bitcoinj.core.FilteredBlock
+import org.bitcoinj.core.Peer
+import org.bitcoinj.core.listeners.BlocksDownloadedEventListener
+import provider.btc.BtcRegisteredAddressesProvider
+import sidechain.SideChainEvent
+
+private const val DAY_MILLIS = 24 * 60 * 60 * 1000L
+
+/*
+    Listener of Bitcoin blockchain events.
+ */
+class BitcoinBlockChainListener(
+    private val btcRegisteredAddressesProvider: BtcRegisteredAddressesProvider,
+    private val emitter: ObservableEmitter<SideChainEvent.PrimaryBlockChainEvent>,
+    private val confidenceLevel: Int
+) : BlocksDownloadedEventListener {
+
+    override fun onBlocksDownloaded(peer: Peer?, block: Block?, filteredBlock: FilteredBlock?, blocksLeft: Int) {
+        if (block == null) {
+            logger.warn { "Cannot handle null BTC block" }
+            return
+        } else if (block.time.time < System.currentTimeMillis() - DAY_MILLIS) {
+            //We cannot handle too old blocks due to Iroha time restrictions.
```

Comment:
Should we do something in this case?

Reply:
We cannot do anything in this case. At least at d3 side.